bentoml/deployment/sagemaker/__init__.py

```diff
@@ -351,151 +355,163 @@ def apply(self, deployment_pb, yatai_service, prev_deployment=None):
                     bento_version=deployment_spec.bento_version,
                 )
             )
-            if bento_pb.bento.uri.type != BentoUri.LOCAL:
+            if bento_pb.bento.uri.type not in (BentoUri.LOCAL, BentoUri.S3):
                 raise BentoMLException(
-                    'BentoML currently only support local repository'
+                    'BentoML currently not support {} repository'.format(
+                        bento_pb.bento.uri.type
+                    )
                 )
-            else:
-                bento_path = bento_pb.bento.uri.uri
 
-            ensure_deploy_api_name_exists_in_bento(
-                [api.name for api in bento_pb.bento.bento_service_metadata.apis],
-                [sagemaker_config.api_name],
+            return self._apply(
+                deployment_pb,
+                bento_pb,
+                yatai_service,
+                bento_pb.bento.uri.uri,
+                prev_deployment,
             )
 
-            sagemaker_client = boto3.client('sagemaker', sagemaker_config.region)
+        except BentoMLException as error:
+            return ApplyDeploymentResponse(status=exception_to_return_status(error))
 
-            with TempDirectory() as temp_dir:
-                sagemaker_project_dir = os.path.join(
-                    temp_dir, deployment_spec.bento_name
-                )
-                init_sagemaker_project(sagemaker_project_dir, bento_path)
-                ecr_image_path = create_push_docker_image_to_ecr(
-                    sagemaker_config.region,
-                    deployment_spec.bento_name,
-                    deployment_spec.bento_version,
-                    sagemaker_project_dir,
+    def _apply(
+        self, deployment_pb, bento_pb, yatai_service, bento_path, prev_deployment=None
+    ):
+        if loader._is_remote_path(bento_path):
```

Comment:
could you add some unit test for SageMakerDeploymentOperator#apply that covers this case? you can just mock the `loader._resolve_remote_bundle_path` to return a path in the test

Reply:
Yes, will do.